web/pdf_find_controller.js

```diff
@@ -158,13 +159,7 @@ class PDFFindController {
     this._dirtyMatch = false;
     this._findTimeout = null;
 
-    this._firstPagePromise = new Promise((resolve) => {
-      const eventBus = this._eventBus;
-      eventBus.on('pagesinit', function onPagesInit() {
-        eventBus.off('pagesinit', onPagesInit);
-        resolve();
-      });
-    });
+    this._firstPageCapability = createPromiseCapability();
```

Comment:
I'm wondering if `_firstPageCapability` is a good name now that we're not actually waiting until the `pagesinit` event is emitted. Maybe `_initializationCapability` or something similar would be better; what do you think?

Reply:
> I'm wondering if `_firstPageCapability` is a good name now that we're not actually waiting until the `pagesinit` event is emitted.

The answer is probably both yes and no at this point, depending on your perspective; naming thing is hard :-)

The name is still very much relevant for the default use-case, i.e. the viewer/components, since `PDFFindController.setDocument` is still called *after* the first page has been resolved in `BaseViewer.setDocument`.
However, for the general case where `PDFFindController` could now be used standalone from the viewer, the name is obviously somewhat strange.

Given that it's tracking internal state anyway, I don't know if the name matter too much though and keeping the current one is also more inline with the default viewer use-case.

Reply:
> I'm wondering if `_firstPageCapability` is a good name now that we're not actually waiting until the `pagesinit` event is emitted.

The answer is probably both yes and no at this point, depending on your perspective; naming things is hard :-)

The name is still very much relevant for the default use-case, i.e. the viewer/components, since `PDFFindController.setDocument` is still called *after* the first page has been resolved in `BaseViewer.setDocument`.
However, for the general case where `PDFFindController` could now be used standalone from the viewer, the name is obviously somewhat strange.

Given that it's tracking internal state anyway, I don't know if the name matter too much though and keeping the current one is also more inline with the default viewer use-case.